build.sbt

```diff
@@ -1,66 +1,81 @@
-ThisBuild / version := "0.1-SNAPSHOT"
-ThisBuild / scalaVersion := "2.12.18"
+import Build._
 
-assembly / assemblyMergeStrategy := {
-  case PathList("META-INF", "services", _*) => MergeStrategy.filterDistinctLines
-  case PathList("META-INF", _*) => MergeStrategy.discard
-  case PathList("module-info.class") => MergeStrategy.discard
-  case PathList("org", "apache", "commons", "logging", _*) =>
-    MergeStrategy.first
-  case x =>
-    val oldStrategy = (assembly / assemblyMergeStrategy).value
-    oldStrategy(x)
-}
+// reload sbt when build definition changes
+Global / onChangedBuildSource := ReloadOnSourceChanges
 
-assembly / assemblyShadeRules := Seq(
-  ShadeRule.rename(
-    "com.google.common.**" -> "shaded.com.google.common.@1"
-  ).inAll
+inThisBuild(
+  Seq(
+    scalaVersion := V.scala212,
+    organization := "com.worksap",
+    homepage := Some(url("https://github.com/WorksApplications/uzushio")),
+    versionScheme := Some("early-semver"),
+    developers := List(
+      Developer(
+        "eiennohito",
+        "Arseny Tolmachev",
+        "arseny@kotonoha.ws",
+        url("https://github.com/eiennohito")
+      )
+    )
+  )
 )
 
-val spark = Seq(
-  "org.apache.spark" %% "spark-sql" % "3.3.2",
-  "org.apache.spark" %% "spark-mllib" % "3.3.2",
+lazy val commonSettings = Seq(
+  crossScalaVersions := Seq(V.scala212, V.scala213),
+  scalacOptions ++= Seq(
+    "-feature",
+    "-deprecation",
+    "-unchecked",
+    "-encoding",
+    "utf-8"
+  ),
+  javacOptions ++= Seq(
+    "-encoding",
+    "utf8",
+    "-Xlint:all",
+    "-source",
+    "1.8",
+    "-target",
+    "1.8"
+  )
 )
 
 lazy val root = (project in file("."))
-  .dependsOn(lib)
+  .aggregate(
```

Comment:
This aggregates submodules' tasks in root module. Now you can run submodule tasks(comile, test) in parallel.